src/commands/GameIntegration/fortnite.ts

```diff
@@ -0,0 +1,129 @@
+import { SkyraCommand } from '@lib/structures/SkyraCommand';
+import { UserRichDisplay } from '@lib/structures/UserRichDisplay';
+import { TOKENS } from '@root/config';
+import { BrandingColors } from '@utils/constants';
+import { Fortnite } from '@utils/GameIntegration/Fortnite';
+import { fetch, FetchResultTypes, getColor } from '@utils/util';
+import { MessageEmbed } from 'discord.js';
+import { CommandStore, KlasaMessage } from 'klasa';
+
+export default class extends SkyraCommand {
+
+	private apiBaseUrl = 'https://api.fortnitetracker.com/v1/profile/';
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			cooldown: 10,
+			description: language => language.tget('COMMAND_FORTNITE_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_FORTNITE_EXTENDED'),
+			requiredPermissions: ['EMBED_LINKS'],
+			usage: '<xbox|psn|pc:default> <user:...string>',
+			usageDelim: ' '
+		});
+	}
+
+	public async run(message: KlasaMessage, [platform, user]: [platform, string]) {
+		const response = await message.sendEmbed(new MessageEmbed()
+			.setDescription(message.language.tget('SYSTEM_LOADING'))
+			.setColor(BrandingColors.Secondary));
+
+		const fortniteUser = await this.fetchAPI(message, user, platform);
+		console.log(fortniteUser);
+
+		await this.buildDisplay(message, fortniteUser).start(response, message.author.id);
+		return response;
+	}
+
+	private async fetchAPI(message: KlasaMessage, user: string, platform: platform) {
+		try {
+			const fortniteUser = await fetch(
+				`${this.apiBaseUrl}/${platform}/${user}`,
+				{ headers: { 'TRN-Api-Key': TOKENS.FORTNITE_KEY } },
+				FetchResultTypes.JSON
+			) as Fortnite.FortniteUser;
+
+			if (!fortniteUser.error) return fortniteUser;
+			throw 'err'; // This gets handled in the catch, no reason to get the proper error message here.
+		} catch {
+			// Either when no user is found (response will have an error message)
+			// Or there was a server fault (no json will be returned)
+			throw message.language.tget('COMMAND_FORTNITE_NO_USER');
+		}
+	}
```

Comment:
If possible this should be a "fetchUser" (or something along these lines) helper function in the GameIntergration util directory.

Reply:
I disagree since it's such a small thing. Clash of Clans also doesn't have that. FFXIV is unique in having it since there's 3 calls and then some just to get all the data and everything in an embeddable format.